pkg/driver/utils.go

```diff
@@ -106,3 +107,29 @@ func getCapacity(cvc *apisv1.CStorVolumeConfig) string {
 	qCap := cvc.Spec.Capacity[corev1.ResourceStorage]
 	return qCap.String()
 }
+
+func getVolumeCondition(vol *apisv1.CStorVolume) *csi.VolumeCondition {
+	condition := &csi.VolumeCondition{}
+	if vol.Status.Phase != apisv1.CVStatusHealthy {
+		condition.Abnormal = true
+	}
+
+	switch vol.Status.Phase {
+	case apisv1.CVStatusHealthy:
+		condition.Message = "Volume status is Healthy"
+
+	case apisv1.CVStatusInit:
+		condition.Message = "Volume is getting initialized, replicas are not yet connected to the target"
+
+	case apisv1.CVStatusOffline:
+		condition.Message = fmt.Sprintf("Volume status is offline, No replicas are connected to target")
+
+	case apisv1.CVStatusDegraded:
+		condition.Message = fmt.Sprintf("Volume status is degraded, more then 1 replica is not connected to target")
```

Comment:
Degrade means replicas are under rebuilding state... god to even log the no.of healthy replicas and degraded replicas also
```suggestion
		condition.Message = fmt.Sprintf("Volume status is degraded, quorum no.of replicas are not in healthy state")
```

Reply:
done